url6dc6a3bff8_JasonxJiang_PokemonEvolutionaryAlgorithm_tgz-pJ8-ee_ea_pkg1_EEEA1J8/src/ee/ea/pkg1/pokemon/DeoxysA.java

```diff
@@ -0,0 +1,186 @@
+            natureSpeed = 1.0;
+            IVHP = 31;
+            IVATK = 31;
+            IVDEF = 31;
+            IVSPA = 31;
+            IVSPD = 31;
+            IVSPEED = 31;
+            EVHP = 0;
+            EVATK = 126;
+            EVDEF = 0;
+            EVSPA = 126;
+            EVSDF = 0;
+            EVSPEED = 252;
+            statBoostATK = 1;
+            statBoostDEF = 1;
+            statBoostSPA = 1;
+            statBoostSDF = 1;
+            statBoostSPEED = 1;
+            level = 100;
+            finalHp = (int) ((((IVHP + (2 * baseHealth) + (EVHP / 4)) * level) / 100) + 10);
+            finalAtk = (int) (((((IVATK + (2 * baseAttack) + (EVATK / 4)) * level) / 100) + 5) * natureAtk);
+            finalDef = (int) (((((IVDEF + (2 * baseDefense) + (EVDEF / 4)) * level) / 100) + 5) * natureDef);
+            finalSpa = (int) (((((IVSPA + (2 * baseSpecialAttack) + (EVSPA / 4)) * level) / 100) + 5) * natureSpa);
+            finalSdf = (int) (((((IVSPD + (2 * baseSpecialDefense) + (EVSDF / 4)) * level) / 100) + 5) * natureSdf);
+            finalSpeed = (int) (((((IVSPEED + (2 * baseSpeed) + (EVSPEED / 4)) * level) / 100) + 5) * natureSpeed);
+            name = "Deoxys-A";
+            primary = new Psychic();
+            secondary = new None();
+            moveLibrary = new HashMap<String, Move>();
+            Move m1 = new ExtremeSpeed();
+            Move m2 = new FirePunch();
+            Move m3 = new IceBeam();
+            Move m4 = new PsychoBoost();
+            Move m5 = new SuperPower();
+            Move m6 = new ThunderBolt();
+            Move m7 = new ShadowBall();
+            moveLibrary.put("extreme speed", m1);
+            moveLibrary.put("fire punch", m2);
+            moveLibrary.put("ice beam", m3);
+            moveLibrary.put("psycho boost", m4);
+            moveLibrary.put("superpower", m5);
+            moveLibrary.put("thunderbolt", m6);
+            moveLibrary.put("shadow ball", m7);
+            potentials = new ArrayList<String>();
+            potentials.add("extreme speed");
+            potentials.add("fire punch");
+            potentials.add("ice beam");
+            potentials.add("psycho boost");
+            potentials.add("superpower");
+            potentials.add("thunderbolt");
+            potentials.add("shadow ball");
+            circumvent = new ArrayList<String>();
+            battleReady = new HashMap<String, Move>();
+            for (int i = 0; i < 4; i++) {
+                int randomIndex = (int) (Math.random() * potentials.size());
+                String key = potentials.get(randomIndex);
+                while (circumvent.contains(key)) {
+                    int nextRandom = (int) (Math.random() * potentials.size());
+                    key = potentials.get(nextRandom);
+                }
+                battleReady.put(key, moveLibrary.get(key));
+                circumvent.add(key);
+                //potentials.remove(randomIndex);
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 93:103, 172:182.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.